### PR TITLE
cmd/cloner: add regression test for slice nil/empty semantics

### PR DESCRIPTION
**Pull request by raggi**
We had a misstep with the semantics when applying an optimization that showed up in the roll into corp. This test ensures that case and related cases must be retained.

Updates #9410
Updates #9601
Signed-off-by: James Tucker <james@tailscale.com>

### CODE REVIEW

File: cmd/cloner/cloner_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+			in:   nil,
+		},
+		{
+			name: "zero",
+			in:   &clonerex.SliceContianer{},
+		},
+		{
+			name: "empty",
+			in: &clonerex.SliceContianer{
+				Slice: []*int{},
```
Comment (maisem) [inline cmd/cloner/cloner_test.go:28]: add non-empty too?

**Review by maisem — APPROVED** [review]
Thank you!

**Review by maisem — APPROVED** [review]
